ui/app/components/auth-jwt.js

```diff
@@ -0,0 +1,163 @@
+import Ember from 'ember';
+import { inject as service } from '@ember/service';
+import Component from './outer-html';
+import { next, later } from '@ember/runloop';
+import { task, timeout, waitForEvent } from 'ember-concurrency';
+import { computed } from '@ember/object';
+
+const WAIT_TIME = Ember.testing ? 0 : 500;
+const ERROR_WINDOW_CLOSED =
+  'The provider window was closed before authentication was complete.  Please click Sign In to try again.';
+const ERROR_MISSING_PARAMS =
+  'The callback from the provider did not supply all of the required parameters.  Please click Sign In to try again. If the problem persists, you may want to contact your administrator.';
+
+export { ERROR_WINDOW_CLOSED, ERROR_MISSING_PARAMS };
+
+export default Component.extend({
+  store: service(),
+  selectedAuthPath: null,
+  roleName: null,
+  role: null,
+  onRoleName() {},
+  onLoading() {},
+  onError() {},
+  onToken() {},
+  onNamespace() {},
+
+  didReceiveAttrs() {
+    next(() => {
+      let { oldSelectedAuthPath, selectedAuthPath } = this;
+      if (oldSelectedAuthPath !== selectedAuthPath) {
+        this.set('role', null);
+        this.onRoleName(null);
+        this.fetchRole.perform(null, { nodebounce: true });
```

Comment:
lol oh no, I'll add some casing in there

Reply:
Is `nodedebounce` a custom property then? I assumed it was something you couldn't touch. Ah just seen where you are using it. 

How about `{debounce: false}`? Maybe another personal thing, but I generally try to avoid negative boolean property/method names so you don't get double negatives.